src/main/java/edu/wpi/grip/core/SocketHints.java

```diff
@@ -0,0 +1,177 @@
+package edu.wpi.grip.core;
+
+
+import org.bytedeco.javacpp.opencv_core.Mat;
+import org.bytedeco.javacpp.opencv_core.Point;
+import org.bytedeco.javacpp.opencv_core.Size;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
+
+
+/**
+ * Commonly used {@link SocketHint SocketHints}
+ */
+public final class SocketHints {
+
+    /**
+     * Provides some default domains
+     */
+    public enum Domain {
```

Comment:
This class is not necessary, and is actually IMO harmful.

If you want a domain of positive integers, you can use `(..., 1, Integer.MAX_VALUE);`.  This is more concise than using this enum, and also more consistent - it makes it so you use the same semantics to set a range of 1 to 100 as 1 to `Integer.MAX_VALUE`, which I think would cause less confusion among people writing python scripts